hangman/README.md

```diff
@@ -0,0 +1,96 @@
+# Hangman - Coding Task
+
+## Overview
+**Hangman** is a classic word-guessing game where players attempt to guess a hidden word by suggesting letters. If the player makes too many incorrect guesses, they lose the game.
+
+## Task Description
+The goal is to implement a **text-based** version of Hangman. The game should:
+- Select a **random word** from a predefined list.
+- Display the **partially guessed word**, replacing unknown letters with `_`.
+- Allow the player to **guess one letter at a time**.
+- Keep track of **incorrect guesses** and limit them to a set number (e.g., 6).
+- End the game when the player either **guesses the word correctly** or **exceeds the maximum wrong guesses**.
+
+## Example Game Flow
+
+```
+# Game starts
+Hidden word: _ _ _ _ _  
+Incorrect guesses left: 6  
+
+> Guess a letter: e
+
+Hidden word: _ e _ _ _  
+Incorrect guesses left: 6  
+
+> Guess a letter: x
+
+Hidden word: _ e _ _ _  
+Incorrect guesses left: 5  
+
+> Guess a letter: l
+
+Hidden word: _ e l l _  
+Incorrect guesses left: 5  
+
+> Guess a letter: h
+
+Hidden word: h e l l _  
+Incorrect guesses left: 5  
+
+> Guess a letter: l (already guessed)
+
+Invalid input. Try again.
+
+> Guess a letter: o
+
+Hidden word: h e l l o  
+Congratulations! You guessed the word.
+```
+
+## Requirements
+- The game should be **playable via the command line**.
```

Comment:
Do we maybe want to move down in the list of requirements to avoid people focusing on the CLI first rather than the logic of the game? Granted though it shouldn't take long if they start with this but it might be an unnecessary distraction at the beginning?